### PR TITLE
Minor black-engine typing improvement

### PR DESCRIPTION
**Pull request by Dunnings**
Updated typings to be more in line with [source](https://github.com/MassiveHeights/Black/). 

- getSystem / hasSystem needed generic Type typings
- Some callbacks were missing parameters

### CODE REVIEW

**Comment by Dunnings** [comment]
@peterblazejewicz @amcasey any chance one of you guys could take a real quick look?

File: types/black-engine/Engine.d.ts [hunk 1]
```diff
@@ -47,9 +47,9 @@ export class Engine extends MessageDispatcher {
     private __bootStage;
     private __onVisibilityChangeFallback;
     __onVisibilityChange(): void;
-    hasSystem(systemTypeName: () => void): boolean;
+    hasSystem(systemTypeName: any): boolean;
```
Comment (amcasey) [inline types/black-engine/Engine.d.ts:50]: How come this one didn't become `(...params: any[]) => void` like the other callbacks?

File: types/black-engine/Engine.d.ts [hunk 2]
```diff
@@ -47,9 +47,9 @@ export class Engine extends MessageDispatcher {
...
     private __addSystem;
-    getSystem(typeName: () => void): System | null;
+    getSystem<T>(typeName: T): T | null;
```
Comment (amcasey) [inline types/black-engine/Engine.d.ts:52]: It seems strange to me that the `typeName` and the `System` would have the same type.

**Review by amcasey — APPROVED** [review]
For future reference, the maintainers rotate from week to week, so it's generally preferable not to tag them directly - whoever's on will find your PR through the normal workflow.

File: types/black-engine/Engine.d.ts (hunk 1 above)
Comment (Dunnings) [reply types/black-engine/Engine.d.ts:50]: That's because it's not supposed to be a callback, it's a System type

**Comment by Dunnings** [comment]
Ready to merge

**Pull request merged**